partiql-tests-data/eval/primitives/operators/overlaps.ion

```diff
@@ -0,0 +1,779 @@
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with non-overlapping date-interval period",
+    statement:"[DATE '2023-01-01', INTERVAL '10' DAY] OVERLAPS [DATE '2023-01-15', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:false
+    }
+  },
+  {
+    name:"overlaps with touching date-interval period",
+    statement:"[DATE '2023-01-01', INTERVAL '14' DAY] OVERLAPS [DATE '2023-01-15', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:false
+    }
+  },
+  {
+    name:"overlaps with both periods using intervals",
+    statement:"[DATE '2023-01-01', INTERVAL '30' DAY] OVERLAPS [DATE '2023-01-15', INTERVAL '30' DAY]",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with negative interval",
+    statement:"[DATE '2023-01-31', INTERVAL '-30' DAY] OVERLAPS [DATE '2023-01-15', DATE '2023-02-15']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:true
+    }
+  },
+  {
+    name:"overlaps with wrong collection size (first period has 1 element)",
+    statement:"[DATE '2023-01-01'] OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with wrong collection size (first period has 1 element)",
+    statement:"[DATE '2023-01-01'] OVERLAPS [DATE '2023-01-01', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
+  {
+    name:"overlaps with wrong collection size (second period has 3 elements)",
+    statement:"[DATE '2023-01-01', DATE '2023-01-31'] OVERLAPS [DATE '2023-01-01', DATE '2023-01-15', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with wrong collection size (second period has 3 elements)",
+    statement:"[DATE '2023-01-01', DATE '2023-01-31'] OVERLAPS [DATE '2023-01-01', DATE '2023-01-15', DATE '2023-01-31']",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
+  {
+    name:"overlaps with empty collections",
+    statement:"[] OVERLAPS []",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce
+      ],
+      output:null
+    }
+  },
+  {
+    name:"overlaps with empty collections",
+    statement:"[] OVERLAPS []",
+    assert:{
+      result:EvaluationFail,
+      evalMode:[
+        EvalModeError
+      ]
+    }
+  },
```

Comment:
If the behavior differs between eval modes, you can define the `assert` block as a list to contain the assertions for both eval modes. E.g. for abs - https://github.com/partiql/partiql-tests/blob/f7677f297a1a9754385bdd7f06136a5f267eed90/partiql-tests-data/eval/primitives/functions/abs.ion#L98-L110.